src/traversals/greedy_color.jl

```diff
@@ -0,0 +1,155 @@
+"""
+    struct coloring{T}
+
+Store number of colors used and mapping from vertex to color
+"""
+struct coloring{T<:Integer}
+    num_colors::T
+    colors::Vector{T}
+end
+
+best_color(c1::coloring, c2::coloring) = c1.num_colors < c2.num_colors ? c1 : c2
+
+"""
+    perm_greedy_color(g, seq)
+
+Color graph `g` according to an order specified by `seq` using a greedy heuristic.
+seq[i] = v imples that vertex v is the i<sup>th</sup> vertex to be colored.
+"""
+function perm_greedy_color(
+    g::AbstractGraph,
+    seq::Vector{T}
+    ) where T <: Integer 
+
+    nvg::T = nv(g)
+    cols = Vector{T}(nvg)  
+    seen = zeros(Bool, nvg + 1)
+
+    for v in seq
+        seen[v] = true
+        colors_used = zeros(Bool, nvg)
+
+        for w in neighbors(g, v)
+            if seen[w]
+                colors_used[Int(cols[w])] = true
+            end
+        end
+
+     
+        for i in one(T):nvg
+            if colors_used[i] == false
+                cols[v] = i
+                break;
+            end
+        end
+    end
+
+    result = coloring{T}(maximum(cols), cols)
+    return result
+end
+
+"""
+    degree_greedy_color(g)
+
+Color graph `g` iteratively in the descending order of the degree of the vertices.
+"""
+function degree_greedy_color(g::AbstractGraph{U}) where U<:Integer
+    seq::Vector{U} = sortperm(degree(g, vertices(g)) , rev=true)
```

Comment:
`degree(g)` will give you a vector of all degrees indexed by vertex. No need for `vertices(g)` here. Also, you don't need to type assert `seq`.

Reply:
Removing type assert gives an error because sortperm outputs Array{Int64}

Reply:
If you want to convert, do so explicitly:
```
seq = U.(sortperm(degree(g), rev=true)
```